### PR TITLE
Generate sentry source maps at release.

### PR DESCRIPTION
**Pull request by romain-growthbook**
The changes in this PR makes it so that, when deploying a production release, we extract sourcemaps from the docker build, injects sentry debug IDs and create a release on sentry with the sourcemaps.

This should help debugging by being able to pinpoint the exact source location where an error occurs and correlating them with releeases.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -52,6 +52,31 @@ jobs:
...
+      - name: Create Sentry release
+        uses: getsentry/action-release@v1
+        with:
+          environment: 'production'
+          sourcemaps: './packages/back-end/dist'
+          version: ${{ steps.metadata.outputs.sha }}
+        env:
+          SENTRY_AUTH_TOKEN: ${{ secrets.SENTRY_AUTH_TOKEN }}
+          SENTRY_ORG: ${{ secrets.SENTRY_ORG }}
+          SENTRY_PROJECT: ${{ secrets.SENTRY_PROJECT }}
```
Comment (jdorn) [inline .github/workflows/deploy.yml:78]: I would move these steps to the `prod` job below.  The `docker` job is just about uploading the build artifact to Docker Hub.  The `prod` job is about deploying this new image to our Cloud.

A nice side effect of this is that it avoids an unnecessary Docker download.  Both the "Generate sentry sourcemaps" and "Extract static assets" steps need the Docker image, so if they are in the same job, we can share the cost.

Comment (romain-growthbook) [reply .github/workflows/deploy.yml:78]: Excellent suggestion thanks.

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -70,8 +70,31 @@ jobs:
...
+      - name: Create Sentry release
+        uses: getsentry/action-release@v1
+        with:
+          environment: 'production'
+          sourcemaps: './back-end-dist'
+          version: ${{ steps.metadata.outputs.sha }}
+        env:
+          SENTRY_AUTH_TOKEN: ${{ secrets.SENTRY_AUTH_TOKEN }}
+          SENTRY_ORG: ${{ secrets.SENTRY_ORG }}
+          SENTRY_PROJECT: ${{ secrets.SENTRY_PROJECT }}
```
Comment (tzjames) [inline .github/workflows/deploy.yml:97]: I don't think these need to be secrets.  We can just hard code them in.

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -70,8 +70,31 @@ jobs:
...
+          npm exec sentry-cli -- sourcemaps inject ./back-end-dist
+
+      - name: Create Sentry release
+        uses: getsentry/action-release@v1
+        with:
+          environment: 'production'
+          sourcemaps: './back-end-dist'
+          version: ${{ steps.metadata.outputs.sha }}
+        env:
+          SENTRY_AUTH_TOKEN: ${{ secrets.SENTRY_AUTH_TOKEN }}
```
Comment (tzjames) [inline .github/workflows/deploy.yml:95]: I think @jdorn will need to add this one to our secrets. Here is where the token can be created https://growthbook.sentry.io/settings/auth-tokens/

**Review by tzjames — COMMENTED** [review]
In order to have the individual error events tagged we need to add it. 
 I actually did it in https://github.com/growthbook/growthbook/pull/3050/files which I will abandon since I also did the sentry github action there which this PR takes care of.

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (romain-growthbook) [reply .github/workflows/deploy.yml]: Done

**Comment by romain-growthbook** [comment]
Thanks for the review. I think that this is ready but will need to be prepped with the secret and monitored during deploy.

**Review by tzjames — APPROVED** [review]
Looks good.  Hope it works.  I like your suggestion to make sure that there is nothing else that is urgent to deploy as it might take a few tries.

**Review by jdorn — APPROVED** [review]

**Pull request merged**